File: spring-web-java/src/main/java/spring/web/java/domain/member/dto/MemberSaveRequest.java
package spring.web.java.domain.member.dto;

import jakarta.validation.constraints.NotNull;
import spring.web.java.domain.member.Member;

public record MemberSaveRequest(
	@NotNull
	String account,
	@NotNull
	String password,
	@NotNull
	String name,
	@NotNull
	Member.Role role,
	@NotNull
	String city,
	@NotNull
	String street,
	@NotNull
	String zipcode
) {
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/member/dto/MemberSaveRequest.kt
package spring.web.kotlin.domain.member.dto

import spring.web.kotlin.domain.member.Member


data class MemberSaveRequest(
    val account: String,
    val password: String,
    val name: String,
    val role: Member.Role,
    val city: String,
    val street: String,
    val zipcode: String
)
